File: 2023/Day15/part1.py
f = open("input.txt", "r").read().split(",")


def hash(x: str) -> int:
    h = 0
    for c in x:
        h += ord(c)
        h = (17 * h) % 256
    return h


print(sum([hash(x) for x in f]))

File: 2025/Day01/part1.ts
import * as fs from "fs";
import * as path from "path";

// Read and parse input
const inputPath = path.join(__dirname, "input.txt");
const input = fs.readFileSync(inputPath, "utf-8");

// Parse input into lines (remove empty lines at the end)
const lines = input.trim().split("\n");

// Main solution function
function solve() {
  let sum = 0;
  let position = 50;
  for (const instruction of lines) {
    const direction = instruction.at(0);
    const steps = parseInt(instruction.slice(1));

    switch (direction) {
      case "L":
        position = (position - steps) % 100;
        break;
      case "R":
        position = (position + steps) % 100;
        break;
    }

    if (position === 0) {
      sum++;
    }
  }
  console.log(sum);
}

// Run the solution
solve();
